docs/scripting-element-object.rst

```diff
@@ -0,0 +1,743 @@
+screenshot will be shrunk by the specified amount of pixes.
+
+See more in :ref:`splash-jpeg`.
+
+
+.. _splash-element-visible:
+
+element:visible
+---------------
+
+Check whether the element is visible.
+
+**Signature:** ``visible = element:visible()``
+
+**Returns:** ``visible`` indicated whether
+the element is visible.
+
+**Async:** no.
+
+
+.. _splash-element-text:
+
+element:text
+------------
+
+Fetch a text information from the element
+
+**Signature:** ``text = element:text()``
+
+**Returns:** ``text`` is a text content
+of the element.
+
+**Async:** no.
+
+It tries to return the trimmed value of the following JavaScript ``Node`` properties:
+
+* textContent
+* innerText
+* value
+
+If all of them are empty an empty string is returned.
+
+
+.. _splash-element-info:
+
+element:info
+------------
+
+Get useful information about the element.
+
+**Signature:** ``info = element:info()``
+
+**Returns:** ``info`` is a table with element info.
+
+**Async:** no.
+
+Info is a table with the following fields:
+
+* nodeName - node name in a lower case (e.g. *h1*)
+* attributes - table with attributes names and its values
+* tag - html string representation of the element
+* html - inner html of the element
+* text - inner text of the element
+* x - x coordinate of the element
+* y - y coordinate of the element
+* width - width of the element
+* height - height of the element
+* visible - flag representing if the element is visible
+
+
+.. _splash-element-field-value:
+
+element:field_value
+-------------------
+
+Get value of the field element (input, select, textarea, button).
+
+**Signature:** ``ok, value = element:field_value()``
+
+**Returns:** ``ok, value`` pair. If ``ok`` is nil then error happened during the function call;
+``value`` provides an information about error type; otherwise ``info`` is a value of the
+element.
+
+**Async:** no.
+
+This methods in the following way:
+
+    - if the element type is ``select``:
+        - if the ``multiple`` attribute is ``true`` it returns a *table* with the selected values;
+        - otherwise it returns the value of the select;
+    - if the element has attribute ``type="radio"``:
+        - if it's checked returns its value;
+        - other it returns ``nil``
+    - if the element has attribute ``type="checkbox"`` it returns *bool* value
+    - otherwise it returns the value of the ``value`` attribute or *empty string* if it doesn't exist
+
+
+.. _splash-element-form-values:
+
+element:form_values
+-------------------
+
+Return a table with form values if the element type is *form*
+
+**Signature:** ``values, reason = element:form_values()``
+
+**Returns:** ``values, reason`` pair. If ``values`` is nil then error happened during the function call
+or node type is not *form*; ``reason`` provides an information about error type; otherwise
+``values`` is a table with element names as keys and values as values.
```

Comment:
I just realized that this data format won't work if there are several elements with the same name. Browsers send all key-value pairs in this case. This is a common pattern in PHP world:

```
<input name="foo[]" value="bar">
<input name="foo[]" value="egg">
<input name="foo[]" value="spam">
```

-- php even decodes the submission above to a PHP array.

Having a single value per key is more common, but I think we should support multiple values as well, because it is a perfectly standard feature. What do you think about adding `multi` flag to this function? When true, values will be arrays, not bare values. It'd be nice to have it also supported in `element:fill`, but it can be more tricky; maybe that's fine to just document this gotcha for `element:fill`.